modules/gdscript/gdscript_parser.cpp

```diff
@@ -2888,7 +2888,7 @@ void GDScriptParser::_parse_block(BlockNode *p_block, bool p_static) {
 				lv->assign = assigned;
 
 				if (!_end_statement()) {
-					_set_error("Expected end of statement (\"var\").");
+					_set_error(String() + "Expected end of statement (\"var\"), got " + tokenizer->get_token_name(tokenizer->get_token()) + ":" + tokenizer->get_token_identifier() + " instead");
```

Comment:
You can use `vformat()` instead:
```
vformat("Expected end of statement (\"var\"), got %s:%s instead.", tokenizer->get_token_name(tokenizer->get_token()), tokenizer->get_token_identifier())
```

Also do not remove the final dot, error messages are proper sentences that end with a dot.

Reply:
What would be `tokenizer->get_token_identifier()` in this context? In your screenshot it seems to be an empty string.

Reply:
`var my_var = self undefined_idf`
`tokenizer->get_token()` prints `Identifier`
`tokenizer->get_token_identifier()` prints `undefined_idf`
but most tokens doesn't have identifier

I just copied:
https://github.com/godotengine/godot/blob/1500e40ccf88b0bca34667e3113e4d6c18b3de29/modules/gdscript/gdscript_parser.cpp#L5232

Reply:
![error-message](https://user-images.githubusercontent.com/41085900/76305364-07727680-62eb-11ea-943a-79f71f98c074.JPG)

Reply:
any changes in format for the error message ?
@akien-mga

Reply:
any changes in format for the error message ?
@akien-mga

Reply:
Seems good :)